packages/lite-insight/src/attribution/multiDimension.ts

```diff
@@ -0,0 +1,136 @@
+import { Datum } from '../interface';
+
+type disassembleInfoSub = {
+  startValue: number;
+  endValue: number;
+  diff: number;
+  contribution: number;
+};
+
+type disassembleInfo = {
+  total: number;
+  detail: Object;
+};
+
+const enumerateAllDimensionCombinationsByDFS = (
+  item: Datum,
+  index: number,
+  dimensions: string[],
+  result: Object,
+  stack: string[],
+  measure: string,
+  baseDimension: string,
+  basePointLeft: string | number,
+  basePointRight: string | number
+) => {
+  if (index === dimensions.length) {
+    /** termination condition of the recursion algorithm */
+    const resultTemp = JSON.parse(JSON.stringify(result));
+    const dimName = stack.join('-');
+    if (!Object.prototype.hasOwnProperty.call(resultTemp, dimName)) {
+      const dimInfo: disassembleInfo = {
+        total: 0,
+        detail: {},
+      };
+      resultTemp[dimName] = dimInfo;
+    }
+
+    const dimValues: string[] = new Array(stack.length);
+    for (let i = 0; i < stack.length; i += 1) {
+      dimValues[i] = item[stack[i]] as string;
+    }
+    const dimValName = dimValues.join('-');
+    if (!Object.prototype.hasOwnProperty.call(resultTemp[dimName].detail, dimValName)) {
+      const dimInfoSub: disassembleInfoSub = {
+        startValue: 0,
+        endValue: 0,
+        diff: 0,
+        contribution: 0,
+      };
+      resultTemp[dimName].detail[dimValName] = dimInfoSub;
+    }
+
+    const tempDimInfoSub: disassembleInfoSub = resultTemp[dimName].detail[dimValName];
+    resultTemp[dimName].total -= tempDimInfoSub.diff;
+    tempDimInfoSub.startValue += item[baseDimension] === basePointLeft ? (item[measure] as number) : 0;
+    tempDimInfoSub.endValue += item[baseDimension] === basePointRight ? (item[measure] as number) : 0;
+
+    if (!(tempDimInfoSub.startValue === 0 && tempDimInfoSub.endValue === 0)) {
+      tempDimInfoSub.diff = tempDimInfoSub.endValue - tempDimInfoSub.startValue;
+    }
+    resultTemp[dimName].total += tempDimInfoSub.diff;
+    return resultTemp;
+  }
+
+  /** Case1: dimensions[index] is included */
+  stack.push(dimensions[index]);
+  const resultFirst = enumerateAllDimensionCombinationsByDFS(
+    item,
+    index + 1,
+    dimensions,
+    result,
+    stack,
+    measure,
+    baseDimension,
+    basePointLeft,
+    basePointRight
+  );
+  stack.pop();
+
+  /** Case2: dimensions[index] is not included */
+  const resultSecond = enumerateAllDimensionCombinationsByDFS(
+    item,
+    index + 1,
+    dimensions,
+    resultFirst,
+    stack,
+    measure,
+    baseDimension,
+    basePointLeft,
+    basePointRight
+  );
+
+  return resultSecond;
+};
+
+export const attributeSingleMeasure2MultiDimension = (
+  sourceData: Datum[],
+  dimensions: string[],
+  measure: string,
+  baseDimension: string,
+  basePointLeft: string | number,
+  basePointRight: string | number
+) => {
```

Comment:
1. Same as above. Better to add notes, wrap the parameters and return type declaration.
2. Parameter names seem to be kind of confused: measure --> measureName? / measureId? , what is "basePointLeft" ?

Reply:
fixed.